include/libzfs.h

```diff
@@ -732,7 +732,7 @@ extern int zfs_unshare_nfs(zfs_handle_t *, const char *);
 extern int zfs_unshare_smb(zfs_handle_t *, const char *);
 extern int zfs_unshareall_nfs(zfs_handle_t *);
 extern int zfs_unshareall_smb(zfs_handle_t *);
-extern int zfs_unshareall_bypath(zfs_handle_t *, const char *);
+extern int zfs_unshareall_bypath(zfs_handle_t *, const char *, const char *);
```

Comment:
This is a public function which is part of the API so rather than change the interface why don't you add a new one for this purpose.  Although I suspect your right and no one is depending on this interface.